Lib/test/test_importlib/test_util.py

```diff
@@ -758,6 +760,94 @@ def test_source_from_cache_path_like_arg(self):
         expect = os.path.join('foo', 'bar', 'baz', 'qux.py')
         self.assertEqual(self.util.source_from_cache(path), expect)
 
+    @contextlib.contextmanager
+    def bytecode_path(self, path):
+        # Patch directly in globals of the functions under test, because we have
+        # no direct access to the right _bootstrap_external module to patch
+
+        # cache_from_source and source_from_cache come from the same module,
+        # thus they share the same globals dict, so we only have to patch it
+        # once
+        _globals = self.util.cache_from_source.__globals__
+
+        _orig_path = _globals['bytecode_path']
+        _globals['bytecode_path'] = path
+        try:
+            yield
+        finally:
+            _globals['bytecode_path'] = _orig_path
+
+    @unittest.skipUnless(sys.implementation.cache_tag is not None,
```

Comment:
Is the following equivalent?

    @unittest.skipIf(sys.implementation.cache_tag is None, ...)

If so, considering do that.  skipUnless drives me nuts. :)